azure-python/__main__.py

```diff
@@ -0,0 +1,30 @@
+from pulumi_policy import (
+    EnforcementLevel,
+    PolicyPack,
+    ReportViolation,
+    ResourceValidationArgs,
+    ResourceValidationPolicy,
+)
+
+def storage_container_no_public_read_validator(args: ResourceValidationArgs, report_violation: ReportViolation):
+    if args.resource_type == "azure:storage/container:Container" and "containerAccessType" in args.props:
+        access_type = args.props["containerAccessType"]
+        if access_type == "blob" or access_type == "container":
+            report_violation(
+                "Azure Storage Container must not have blob or container access set. " +
+                "Read more about read access here: " +
+                "https://docs.microsoft.com/en-us/azure/storage/blobs/storage-manage-access-to-resources")
+
+storage_container_no_public_read = ResourceValidationPolicy(
+    name="storage-container-no-public-read",
+    description="Prohibits setting the public permission on Azure Storage Blob Containers.",
+    validate=storage_container_no_public_read_validator,
+)
+
+PolicyPack(
+    name="azure-python",
+    enforcement_level=EnforcementLevel.MANDATORY,
+    policies=[
+        storage_container_no_public_read,
+    ],
+)
```

Comment:
These are totally fine as is - but curious whether you think this or the the class-based style should be our default for templates/examples?

Reply:
Good point. The approach I took here will be familiar to folks who are already familiar with writing policies in TypeScript/JavaScript.

Do you think the class-based style will feel more natural/idiomatic for Python developers? If so, we should go with that.

Here's a comparison of the two supported styles:

```python
def s3_no_public_read_validator(args: ResourceValidationArgs, report_violation: ReportViolation):
    if args.resource_type == "aws:s3/bucket:Bucket" and "acl" in args.props:
        acl = args.props["acl"]
        if acl == "public-read" or acl == "public-read-write":
            report_violation(
                "You cannot set public-read or public-read-write on an S3 bucket. " +
                "Read more about ACLs here: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html")

s3_no_public_read = ResourceValidationPolicy(
    name="s3-no-public-read",
    description="Prohibits setting the publicRead or publicReadWrite permission on AWS S3 buckets.",
    validate=s3_no_public_read_validator,
)

PolicyPack(
    name="aws-python",
    enforcement_level=EnforcementLevel.MANDATORY,
    policies=[
        s3_no_public_read,
    ],
)
```

vs

```python
class S3NoPublicReadPolicy(ResourceValidationPolicy):
    def __init__(self):
        super().__init__(
            name="s3-no-public-read",
            description="Prohibits setting the publicRead or publicReadWrite permission on AWS S3 buckets.")

    def validate(self, args: ResourceValidationArgs, report_violation: ReportViolation):
        if args.resource_type == "aws:s3/bucket:Bucket" and "acl" in args.props:
              acl = args.props["acl"]
              if acl == "public-read" or acl == "public-read-write":
                  report_violation("You cannot set public-read or public-read-write on an S3 bucket.")

PolicyPack(
      name="aws-policy-pack",
      enforcement_level=EnforcementLevel.MANDATORY,
      policies=[
          S3NoPublicReadPolicy(),
      ],
  )
```

Reply:
> more natural/idiomatic for Python developers?

Im not a Python dev.. but generally classes are used for objects that have methods on them, so I am included to use functions here.

Reply:
Sounds good - let's keep the function-based style for now then.  Definitely keeps consistency better, which will make explaining the APIs in documentation much easier.